code/go/0chain.net/smartcontract/dbs/event/provider.go

```diff
@@ -50,3 +71,18 @@ func (p *Provider) BeforeCreate(tx *gorm.DB) (err error) {
 	}
 	return
 }
+
+func (edb *EventDb) updateProvidersHealthCheck(updates *[]dbs.DbHealthCheck, tableName ProviderTable) error {
+	logging.Logger.Info("Running update provider health check with data: ", zap.Any("updates", updates), zap.String("tableName", string(tableName)))
+	updateExpr := map[string]interface{}{
```

Comment:
it is not correct, don't update db this way. Remember, we discussed, that you always has previous healthcheck when you emit event, last time is what in provider entity at the moment, and new one is tx.time. So you can calculate uptime change right there and simply provider instance. You can even store uptime in blobber/sharder MPT entity

Reply:
Fixed